src/main/java/com/epam/controller/ToursController.java

```diff
@@ -53,19 +51,29 @@ public ModelAndView getToursList(RedirectAttributes redirectAttributes) {
         return toursModel;
     }
 
-    @PostMapping("/searchtours")
-    public ModelAndView searchTours(@RequestParam String country, @RequestParam String startDate,
+    @PostMapping("/searchtours/{id}")
+    public ModelAndView searchTours(@PathVariable Integer id, @RequestParam String country, @RequestParam String startDate,
                                     @RequestParam String endDate, @RequestParam String numberOfPeople) {
         ModelAndView toursModel = new ModelAndView();
-        toursModel.setViewName("homepage");
         try {
             LocalDate addStartDate = Validator.getDate(startDate, true);
             LocalDate addEndDate = Validator.getDate(endDate, true);
             toursModel.addObject("hotels", hotelService.getMapOfHotels());
-            toursModel.addObject("list", toursOfferService.searchTours(country, addStartDate, addEndDate));
+            int generalAmount = toursOfferService.amountOfToursSearched(country,addStartDate,addEndDate);
+            int amount = toursOfferService.totalAmountOfRows;
+            List<TourOffer> tours = toursOfferService.searchTours(country, addStartDate, addEndDate,id,amount);
+            toursModel.addObject("list", tours);
+            toursModel.addObject("generalAmount", generalAmount);
+            toursModel.addObject("amount", (generalAmount % amount == 0) ? generalAmount / amount : generalAmount / amount + 1);
+            toursModel.addObject("country", country);
```

Comment:
Can u explain why u need set this attributes like a country and etc.?

Reply:
Sure, because its pagination for search, we still need this attributes to pull proper rows.